src/modules/mavlink/mavlink_log_handler.cpp

```diff
@@ -31,558 +31,498 @@
 {
 	switch (msg->msgid) {
 	case MAVLINK_MSG_ID_LOG_REQUEST_LIST:
-		_log_request_list(msg);
+		handle_log_request_list(msg);
 		break;
 
 	case MAVLINK_MSG_ID_LOG_REQUEST_DATA:
-		_log_request_data(msg);
+		handle_log_request_data(msg);
 		break;
 
-	case MAVLINK_MSG_ID_LOG_ERASE:
-		_log_request_erase(msg);
+	case MAVLINK_MSG_ID_LOG_REQUEST_END:
+		handle_log_request_end(msg);
 		break;
 
-	case MAVLINK_MSG_ID_LOG_REQUEST_END:
-		_log_request_end(msg);
+	case MAVLINK_MSG_ID_LOG_ERASE:
+		handle_log_erase(msg);
 		break;
 	}
 }
 
-//-------------------------------------------------------------------
-void
-MavlinkLogHandler::send()
+void MavlinkLogHandler::state_idle()
 {
-	//-- An arbitrary count of max bytes in one go (one of the two below but never both)
-#define MAX_BYTES_SEND 256 * 1024
-	size_t count = 0;
-
-	//-- Log Entries
-	while (_current_status == LogHandlerState::Listing
-	       && _mavlink.get_free_tx_buf() > MAVLINK_MSG_ID_LOG_ENTRY_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES
-	       && count < MAX_BYTES_SEND) {
-		count += _log_send_listing();
-	}
-
-	//-- Log Data
-	while (_current_status == LogHandlerState::SendingData
-	       && _mavlink.get_free_tx_buf() > MAVLINK_MSG_ID_LOG_DATA_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES
-	       && count < MAX_BYTES_SEND) {
-		count += _log_send_data();
+	if (_current_entry.fp && _file_send_finished) {
+		fclose(_current_entry.fp);
+		_current_entry.fp = nullptr;
+
+		_current_entry.id = 0xffff;
+		_current_entry.time_utc = 0;
+		_current_entry.size_bytes = 0;
+		_current_entry.filepath[0] = 0;
+		_current_entry.offset = 0;
+
+		_entry_request.id = 0xffff;
+		_entry_request.start_offset = 0;
+		_entry_request.byte_count = 0;
 	}
 }
 
-//-------------------------------------------------------------------
-void
-MavlinkLogHandler::_log_request_list(const mavlink_message_t *msg)
+void MavlinkLogHandler::state_listing()
 {
-	mavlink_log_request_list_t request;
-	mavlink_msg_log_request_list_decode(msg, &request);
+	static constexpr uint32_t MAVLINK_PACKET_SIZE = MAVLINK_NUM_NON_PAYLOAD_BYTES + MAVLINK_MSG_ID_LOG_ENTRY_LEN;
 
-	//-- Check for re-requests (data loss) or new request
-	if (_current_status != LogHandlerState::Inactive) {
-		//-- Is this a new request?
-		if (request.start == 0) {
-			_current_status = LogHandlerState::Inactive;
-			_close_and_unlink_files();
-
-		} else {
-			_current_status = LogHandlerState::Idle;
-
-		}
+	if (_mavlink.get_free_tx_buf() <= MAVLINK_PACKET_SIZE) {
+		return;
 	}
 
-	if (_current_status == LogHandlerState::Inactive) {
-		//-- Prepare new request
-
-		_reset_list_helper();
-		_init_list_helper();
-		_current_status = LogHandlerState::Idle;
-	}
+	DIR *dp = opendir(kLogDir);
 
-	if (_log_count) {
-		//-- Define (and clamp) range
-		_next_entry = request.start < _log_count ? request.start :
-			      _log_count - 1;
-		_last_entry = request.end   < _log_count ? request.end :
-			      _log_count - 1;
+	if (!dp) {
+		PX4_WARN("No logs available");
```

Comment:
Why is this a warning?

Maybe a lot of these messages should move back to the previous verbose mechanism, or use PX4_DEBUG instead?